### PR TITLE
Update Version and Changelog

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is to update the Changelog, Version Number of the SDK Generator for the preview release. 

@joheredi @deyaaeldeen @ramya-rao-a Please review and approve.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## 1.0.0-preview.1 (2021-05-28)
+
+- Initial preview release of the v6 version of the Autorest Typescript/Javascript SDK Generator.
```
Comment (deyaaeldeen) [inline CHANGELOG.md:3]: v6 or v1?

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
         "@azure/core-http": "^1.2.1-alpha.20201203.2",
```
Comment (sarangan12) [inline package.json:38]: @joheredi @ramya-rao-a Is it Ok to release a preview version of SDK generator with alpha version of core-http? I think this also should be updated to the latest version. Any thoughts???

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+- This first preview includes:
+  - v2 version of the core packages are supported by default.
```
Comment (deyaaeldeen) [inline CHANGELOG.md:5]:
```suggestion
  - core v2 packages are supported by default.
```

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@autorest/typescript",
-    "version": "6.0.0",
+    "version": "1.0.0-preview.1",
```
Comment (deyaaeldeen) [inline package.json:3]:
```suggestion
    "version": "1.0.0-beta.1",
```

File: package.json (hunk 2 above)
Comment (deyaaeldeen) [reply package.json:38]: this is a pretty old version from last year. I think we should update it to a GA one.

File: CHANGELOG.md (hunk 1 above)
Comment (deyaaeldeen) [inline CHANGELOG.md:3]: I think we should mention that this package is a rebranding of the older @microsoft one. cc @ramya-rao-a

File: package.json (hunk 2 above)
Comment (joheredi) [reply package.json]: Yes, we should change this to ^1.2.4

Comment (sarangan12) [reply package.json]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (sarangan12) [reply CHANGELOG.md]: Changed it

Comment (sarangan12) [reply CHANGELOG.md]: Added that line

**Review by joheredi — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+## 1.0.0-beta.1 (2021-05-28)
+
+- Initial preview release of the the Autorest Typescript/Javascript SDK Generator. This package replaces the older [@microsoft.azure/autorest.typescript](https://www.npmjs.com/package/@microsoft.azure/autorest.typescript) package.
```
Comment (deyaaeldeen) [inline CHANGELOG.md:3]:
```suggestion
- Initial preview release of the Autorest Typescript/Javascript SDK Generator. This package replaces the older [@microsoft.azure/autorest.typescript](https://www.npmjs.com/package/@microsoft.azure/autorest.typescript) package.
```

File: package.json [hunk 6]
```diff
@@ -35,10 +35,10 @@
         "@autorest/extension-base": "^3.2.1",
         "@azure-tools/codegen": "^2.5.294",
         "@azure/core-client": "^1.1.2",
-        "@azure/core-http": "^1.2.1-alpha.20201203.2",
+        "@azure/core-http": "^1.2.4",
```
Comment (deyaaeldeen) [inline package.json:38]: I think our tests no longer need it, is there anything else that needs it?

**Pull request merged**